File: backend/app/schemas/disclosure.py
from datetime import date, datetime

from pydantic import BaseModel


class DisclosureItem(BaseModel):
    id: int
    ticker: str | None = None
    corp_name: str | None = None
    report_nm: str | None = None
    rcept_no: str
    flr_nm: str | None = None
    rcept_dt: date | None = None
    report_type: str | None = None
    disclosure_url: str | None = None

    model_config = {"from_attributes": True}


class DisclosureListResponse(BaseModel):
    items: list[DisclosureItem]
    total: int

File: frontend/src/types/disclosure.ts
export interface DisclosureItem {
  id: number
  ticker: string | null
  corp_name: string | null
  report_nm: string | null
  rcept_no: string
  flr_nm: string | null
  rcept_dt: string | null
  report_type: string | null
  disclosure_url: string | null
}

export interface DisclosureListResponse {
  items: DisclosureItem[]
  total: number
}
